test/test_nn.py

```diff
@@ -847,6 +847,18 @@ def test_mismatch_shape_conv2d(self):
 
             F.conv2d(x, w)
 
+    def test_conv2d_discontiguous_weight(self):
+        # Test for https://github.com/pytorch/pytorch/issues/55781
+        x = torch.ones(64, 16, 16, 16)
+        weight = torch.arange(0, 1.0, 1 / 2.0 ** 10).reshape(32, 16, 1, 2)[:, :, :, ::2]
+        self.assertFalse(weight.is_contiguous())
+        y = torch.nn.functional.conv2d(x, weight, None)
+        if torch.backends.mkldnn.is_available():
+            with torch.backends.mkldnn.flags(enabled=False):
```

Comment:
Can you leave a comment that nnpack path will be chosen if mkldnn is disabled? Otherwise it's not obvious that this test is related to the issue.